omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,298 @@
+                identifierSource.setDescription(description.toString());
+            }
+            if (username!= null && StringUtils.isNotBlank(username.toString())) { 
+                    identifierSource.setUser(username.toString()); 
+            }
+            if (password != null && StringUtils.isNotBlank(password.toString())) {
+                if(username == null || StringUtils.isBlank(username.toString())) {
+                    errors.add("username");
+                }
+                else {
+                    identifierSource.setPassword(password.toString());
+                }
+            }
+            if(errors.size() > 0) {
+                throw new ValidationException("The values of the following inputs are missing or invalid: " + errors.toString());
+            }
+
+            identifierSource.setIdentifierType(patientIdentifierType);
+            identifierSource.setName(name.toString());
+            identifierSource.setUrl(url.toString());
+            savedIdentifierSource =  save(identifierSource);
+        }
+        else if(identifierSourceType.equals(Identifier_Pool)){
+            IdentifierPool identifierSource = new IdentifierPool();
+            Object batchSize = postBody.get("batchSize");
+            Object minPoolSize = postBody.get("minPoolSize");
+            Object sourceUuid = postBody.get("sourceUuid");
+            Object sequential = postBody.get("sequential");
+            Object refillWithScheduledTask = postBody.get("refillWithScheduledTask");
+
+            if (sourceUuid == null || StringUtils.isBlank(sourceUuid.toString())) {
+                errors.add("identifier souce");
+            }
+            else{
+                IdentifierSource poolIdentifierSource = Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(sourceUuid.toString());
+                if(poolIdentifierSource != null){
+                    identifierSource.setSource(poolIdentifierSource);
+                }else{
+                    errors.add("identifier souce");
+                }
+            }
+            if (description != null && StringUtils.isNotBlank(description.toString())) {
+                identifierSource.setDescription(description.toString());
+            }
+            if (batchSize != null && StringUtils.isNotBlank(batchSize.toString())) {
+                if(StringUtils.isNumeric(batchSize.toString())) {
+                    identifierSource.setBatchSize(Integer.parseInt(batchSize.toString()));
+                }
+            }
+            if (minPoolSize != null && StringUtils.isNotBlank(minPoolSize.toString())) {
+                if(StringUtils.isNumeric(minPoolSize.toString())) {
+                    identifierSource.setMinPoolSize(Integer.parseInt(minPoolSize.toString()));
+                }
+            }
+            if (sequential != null && StringUtils.isNotBlank(sequential.toString())) {
+                if(parseBoolean(sequential.toString()) != null) {
```

Comment:
Why are you calling parseBoolean twice?
Secondly, is it necessary to keep doing object.toString() every time you call parseBoolean?

Reply:
1. The setSequential() and setRefillWithScheduledTask() methods only take boolean values. I have to call parseBoolean again to convert the valid string to Boolean type.
2. Removed the toString()